framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -412,14 +430,75 @@ public ValueQualifierHierarchy(MultiGraphQualifierHierarchy.MultiGraphFactory fa
             super(factory);
         }
 
+        /**
```

Comment:
Can you document why a more precise GLB is needed for StringVal but not for other types?